src/v/CMakeLists.txt

```diff
@@ -3,6 +3,41 @@ include_directories(${CMAKE_CURRENT_BINARY_DIR})
 
 add_compile_options(-Wall)
 
+#
+# When enabled clang-tidy will be invoked along side the compiler, and
+# clang-tidy warnings will be reported as errors.
+#
+option(REDPANDA_RUN_CLANG_TIDY "Enable clang-tidy checks" OFF)
+if(REDPANDA_RUN_CLANG_TIDY)
```

Comment:
will you also be adding an option to the task file to enable this?

Reply:
yeh. i'll post a vtools PR as soon as this merges